src/library/scala/math/Ordering.scala

```diff
@@ -214,26 +204,59 @@ trait LowPriorityOrderingImplicits {
   * new orderings.
   */
 object Ordering extends LowPriorityOrderingImplicits {
+  private final val reverseSeed  = 41
+  private final val optionSeed   = 43
+  private final val iterableSeed = 47
+
   @inline def apply[T](implicit ord: Ordering[T]) = ord
 
+  /** A reverse ordering */
+  private final class Reverse[T](private val outer: Ordering[T]) extends Ordering[T] {
```

Comment:
Reverse orderings are also potentially optimisable. Should this class be public?

Reply:
(`outer` as well?)

Reply:
e.g.

```scala
def areReverses[T](ord1: Ordering[T], ord2: Ordering[T]): Boolean =
  (ord1.isInstanceOf[Reverse[_]] && ord1.asInstanceOf[Reverse[T]].outer == ord2) ||
  (ord2.isInstanceOf[Reverse[_]] && ord2.asInstanceOf[Reverse[T]].outer == ord1)
```